Importer/CsvImporter.php

```diff
@@ -23,88 +27,237 @@ class CsvImporter extends AbstractImporter
-     * @param string                   $delimiter
+     * @var ConfigurationInterface
+     */
+    protected $configuration;
+
+    /**
+     * @var EntityManager
+     */
+    protected $entityManager;
+
+    /**
+     * @var ImporterErrorHandlerInterface[]
+     */
+    protected $importerErrorHandlers = [];
+
+    /**
+     * @var InterpreterInterface
+     */
+    protected $interpreter;
+
+    /**
+     * @var LogInterface
+     */
+    protected $log;
+
+    /**
+     * @var bool
+     */
+    protected $csvHasHeaders = true;
+
+    /**
+     * @param ConfigurationInterface $configuration
+     * @param InterpreterInterface   $interpreter
+     * @param EntityManager          $entityManager
+     * @param CsvParser              $parser
+     * @param LogInterface           $log
+     * @param string                 $delimiter
      */
     public function __construct(
         ConfigurationInterface $configuration,
         InterpreterInterface $interpreter,
         EntityManager $entityManager,
         CsvParser $parser,
+        LogInterface $log,
         $delimiter = ','
-    )
-    {
+    ) {
         $this->parser = $parser;
-        parent::__construct($configuration, $interpreter, $entityManager);
         $this->delimiter = $delimiter;
+        $this->configuration = $configuration;
+        $this->entityManager = $entityManager;
+        $this->interpreter = $interpreter;
+        $this->log = $log;
+    }
+
+    /**
+     * @return boolean
+     */
+    public function csvHasHeaders()
+    {
+        return $this->csvHasHeaders;
+    }
+
+    /**
+     * @param boolean $csvHasHeaders
+     */
+    public function setCsvHasHeaders($csvHasHeaders)
+    {
+        $this->csvHasHeaders = $csvHasHeaders;
+    }
+
+    /**
+     * @param string $delimiter
+     */
+    public function setDelimiter($delimiter)
+    {
+        $this->delimiter = $delimiter;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function registerInterpreterErrorHandler(InterpreterErrorHandlerInterface $lineErrorHandler)
+    {
+        $this->interpreter->registerErrorHandler($lineErrorHandler);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function registerImporterErrorHandler(ImporterErrorHandlerInterface $fileErrorHandler)
+    {
+        $this->importerErrorHandlers[] = $fileErrorHandler;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getConfiguration()
+    {
+        return $this->configuration;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getLog()
+    {
+        return $this->log;
     }
 
     /**
-     * @param string $filename
-     * @param bool   $hasHeaders
+     * @param string $csv
      * @param bool   $dryRun
-     *
-     * @return array|null|object[]
-     * @throws DatabaseException
      */
-    public function importFile($filename, $hasHeaders = true, $dryRun = false)
+    public function import($csv, $dryRun = false)
```

Comment:
Technically, this `= false` part breaks compatibility with interface. Interface doesn't have any default value.